vecfixed/src/lib.rs

```diff
@@ -0,0 +1,107 @@
+//noinspection ALL
+/// VecFixed is basically a vector that keep a fixed size. Every time new element is pushed
+/// to the vector, the oldest element is removed and the latest pushed is added to the end.
+#[derive(Default)]
+pub struct VecFixed<const N: usize, T: Default + ToString> {
+    start: usize,
+    buffer: Vec<T>,
```

Comment:
What about using a `VecDeque` here? Since we want to drop from the left and push to the right, it would be better in terms of performances. `rotate_left` for `Vec` is O(n). While it's O(1) for `VecDeque` (in cases like ours where we rotate by 1). Or if we use `VecDeque` we also have `pop_front` if we don't want to rotate, so we could do 
```rust
pop_front();
push_back(elem);
```

Reply:
Does it? We can instantiate a `VecDeque::with_capacity` and then once we are at length == N we pop from the back and we push to the front

Reply:
Ok I've checked the source code:

![image](https://github.com/RIP-Comm/clementine/assets/5256712/3c6938a0-a815-4eb8-853d-fb95bb71a6bf)

So it only grows when it's full, in our case it should not be the case

Reply:
Yes but you still need to handle manually the shift and deletions of elements... More or less is same as this :)

Reply:
![image](https://github.com/RIP-Comm/clementine/assets/5256712/a344f44f-2509-46ba-95a5-26294a7f76c9)

I think in our case it should be O(1) since `mid` for us is constant.

It will probably not change much in terms of performances (or will it?) but since we're writing our own boundedqueue, we can try to optimize it as much as reasonable

Reply:
Yes seems a good improvement, I can change it :)